dev/io.openliberty.microprofile.telemetry.internal.common/src/io/openliberty/microprofile/telemetry/internal/common/info/OpenTelemtryLifecycleManagerImpl.java

```diff
@@ -0,0 +1,240 @@
+import com.ibm.websphere.ras.TraceComponent;
+import com.ibm.ws.container.service.app.deploy.ApplicationInfo;
+import com.ibm.ws.container.service.app.deploy.extended.ExtendedApplicationInfo;
+import com.ibm.ws.container.service.metadata.MetaDataSlotService;
+import com.ibm.ws.container.service.state.ApplicationStateListener;
+import com.ibm.ws.container.service.state.StateChangeException;
+import com.ibm.ws.runtime.metadata.ApplicationMetaData;
+import com.ibm.ws.runtime.metadata.ComponentMetaData;
+import com.ibm.ws.runtime.metadata.MetaDataSlot;
+import com.ibm.ws.threadContext.ComponentMetaDataAccessorImpl;
+import com.ibm.wsspi.kernel.feature.LibertyFeature;
+
+import io.openliberty.checkpoint.spi.CheckpointHook;
+import io.openliberty.checkpoint.spi.CheckpointPhase;
+import io.openliberty.microprofile.telemetry.internal.common.constants.OpenTelemetryConstants;
+import io.openliberty.microprofile.telemetry.internal.interfaces.OpenTelemetryInfoFactory;
+
+@Component(service = { ApplicationStateListener.class, OpenTelemetryLifecycleManager.class }, property = { "service.vendor=IBM", "service.ranking:Integer=1500" })
+public class OpenTelemtryLifecycleManagerImpl implements ApplicationStateListener, OpenTelemetryLifecycleManager {
+
+    private static final TraceComponent tc = Tr.register(OpenTelemtryLifecycleManagerImpl.class);
+
+    private final MetaDataSlot slotForOpenTelemetryInfoHolder;
+    private final OpenTelemetryInfoFactory openTelemetryInfoFactory;
+
+    //These are set in @Activate, and refreshed on checkpoint restore.
+    private boolean isRuntimeEnabled;
+    private LazyInitializer<OpenTelemetryInfo> runtimeInstance;
+
+    @Activate
+    public OpenTelemtryLifecycleManagerImpl(@Reference MetaDataSlotService slotService, @Reference OpenTelemetryInfoFactory openTelemetryInfoFactory,
+                                            @Reference(service = LibertyFeature.class, target = "(ibm.featureName=mpTelemetry-*)") ServiceReference<LibertyFeature> featureRef) {
+        this.openTelemetryInfoFactory = openTelemetryInfoFactory;
+
+        String shortName = (String) featureRef.getProperty("ibm.featureName");
+        //the runtimeMode was added in 2.0
+        if (shortName.startsWith("mpTelemetry-1")) {
+            isRuntimeEnabled = false;
+        } else {
+            HashMap<String, String> propreties = OpenTelemetryPropertiesReader.getRuntimeInstanceTelemetryProperties();
+            isRuntimeEnabled = !!!OpenTelemetryPropertiesReader.checkDisabled(propreties);
+        }
+
+        slotForOpenTelemetryInfoHolder = slotService.reserveMetaDataSlot(ApplicationMetaData.class);
+
+        CheckpointHook checkpointHook = new CheckpointHook() {
+
+            @Override
+            public void restore() {
+
+                HashMap<String, String> propreties = OpenTelemetryPropertiesReader.getRuntimeInstanceTelemetryProperties();
+                isRuntimeEnabled = OpenTelemetryPropertiesReader.checkDisabled(propreties);
```

Comment:
This is problematic if using `mpTelemetry-1` as it may set `isRuntimeEnabled` to `true` when I don't think it should.  I suggest not registering the hook if using  `mpTelemetry-1`

Reply:
This got me thinking and I double checked some stuff. We actually do have a difference in behaviour for instnatOn in app mode too. But I've made it check `mpTelemetry-1` during the restore too.